doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+
+#### Datastream Message (individual aggregation)
+
+Datastream messages for interfaces with individual aggregation have a "v" key and an optional "t" key (which means timestamp) and they look like the followings:
+
+* `{"v": false}`
+
+* `{"v": 16.73}`
+
+* `{"v": 16.73, "t": 1537449422890}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 09 00 00 00 08 76 00 00 00
+```
+
+```
+0000000 10 00 00 00 01 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+```
+0000000 1b 00 00 00 09 74 00 2a 70 20 f7 65 01 00 00 01
+0000020 76 00 7b 14 ae 47 e1 ba 30 40 00
+```
+
+#### Datastream Message (object aggregation)
+
+Datastream messages (for interfaces with object aggregation) have a "v" key, that is a BSON subdocument, and an optional "t" key (which means timestamp) and they look like the following:
+
+* `{"v": {"temp": 25.3123, "hum": 67.112}}`
+
+* `{"v": {"temp": 25.3123, "hum": 67.112}, "t": 1537452514811}`
+
+Timestamps are UTC timestamps (BSON 0x09 type), when not provided reception timestamp is used.
+
+Previous payloads are in order BSON encoded as following hex dumps:
+
+```
+0000000 28 00 00 00 03 76 00 20 00 00 00 01 68 75 6d 00
+0000020 ba 49 0c 02 2b c7 50 40 01 74 65 6d 70 00 72 8a
+0000040 8e e4 f2 4f 39 40 00 00
+```
+
+```
+0000000 33 00 00 00 09 74 00 fb 9d 4f f7 65 01 00 00 03
+0000020 76 00 20 00 00 00 01 68 75 6d 00 ba 49 0c 02 2b
+0000040 c7 50 40 01 74 65 6d 70 00 72 8a 8e e4 f2 4f 39
+0000060 40 00 00
+```
+
+## Minimal Protocol
+
+A device might implement a subset of this protocol if needed. `/control/consumer/properties`, `/control/producer/properties` and `/emptyCache` might be ignored or not implemented if a device has no property interfaces.
+A further simplification might remove any requirement for any introspection message when previously provisioned, but this feature is not supported out of the box.
+
+## Error Handling
+
+A device might be forcefully disconnected due to any kind of error. Devices should wait a random ammount of time before trying to connect again to the broker.
+_session present_ might be also set to false to ensure a clean and consistent state (in that case messages such as _introspection_ and _empty cache_ should published as previously described).
+
+Malformed or unexpected messages are discarded and further actions might be taken.
+
+## Authentication
+In Astarte, every Transport orchestrates its credentials through Pairing. Astarte/VerneMQ authenticates devices using Mutual SSL Autentication - as such, devices use SSL certificates emitted through [Pairing API](050-pairing_mechanism.html) to authenticate against the broker. To achieve this, the device must ensure it is capable of performing http(s) calls to Pairing API to obtain its certificates, performing SSL/X509 operations and of connecting to the MQTT Broker through the use of SSL certificates.
+
```

Comment:
We have to add this paragraph, after discussing with @rbino :

```
## Connecting to the Broker

In the same fashion as Authentication, Pairing provides the client with information about how to connect to the MQTT broker. When invoking relevant [Pairing API's method to gather information about available transports for a device](/api/?urls.primaryName=Pairing%20API#/device/getInfo), if Astarte advertises Astarte MQTT v1, a similar reply will be returned:

```
{
  "data": {
    "version": "string",
    "status": "string",
    "protocols": {
      "astarte_mqtt_v1": {
        "broker_url": "mqtts://broker.astarte.example.com:8883"
      }
    }
  }
}
```

`broker_url` contains a MQTTS (as SSL is always enforced by Astarte MQTT v1) URL the client should connect to.
```